cmd/account/reset.go

```diff
@@ -107,32 +110,63 @@ func (o *resetOptions) run() error {
 		if strings.HasPrefix(secret.Name, o.accountName) {
 			fmt.Fprintln(o.Out, "Deleting secret "+secret.Name)
 			if err := o.kubeCli.Delete(ctx, &secret, &client.DeleteOptions{}); err != nil {
+
 				if apierrors.IsNotFound(err) {
 					continue
 				}
 				return err
 			}
 		}
 	}
-
 	account, err := k8s.GetAWSAccount(ctx, o.kubeCli, o.accountNamespace, o.accountName)
 	if err != nil {
 		return err
 	}
-
+	//get accountID for rest
+	accountId := account.Spec.AwsAccountID
 	// reset fields in spec
 	account.Spec.ClaimLink = ""
 	account.Spec.ClaimLinkNamespace = ""
 	account.Spec.IAMUserSecret = ""
+	//create an awsClient from the credentials in aws-account-operator-credentials
 	if o.resetLegalEntity {
+		awsClient, err := o.getAwsClientFromSecret("aws-account-operator-credentials", "aws-account-operator")
+		if err != nil {
+			return err
+		}
+		//awsClient := *AwsClient
+		//get the rootOU from the awsClient above
+		root, err := awsClient.ListRoots(&organizations.ListRootsInput{})
+		rootId := *root.Roots[0].Id
+		if err != nil {
```

Comment:
Shouldn't we first check for errors since otherwise this call may segfault if `root.Roots` is empty?